internal/database/mysql.go

```diff
@@ -16,172 +17,156 @@ import (
 )
 
 type MySQLDatabase struct {
-	rootDsn      string
+	rootDSN      string
 	dbName       string
 	underlyingDB *sql.DB
 }
 
-func New(dbName string, username string, password string) *MySQLDatabase {
-	mysql := MySQLDatabase{
-		rootDsn:      username + ":" + password + "@tcp(localhost:3306)/",
+func New(dbName string, username string, password string) (mysql *MySQLDatabase) {
+	// Initialized struct to be returned.
+	mysql = &MySQLDatabase{
+		rootDSN:      username + ":" + password + "@tcp(localhost:3306)/",
 		dbName:       dbName,
 		underlyingDB: nil,
 	}
-	initScript, err := os.ReadFile("internal/database/queries/init.sql")
-	if err != nil {
-		log.Fatalln("Error reading init.sql file -", err)
-	}
 
-	// Connect to MySQL root.
-	db, err := sql.Open("mysql", mysql.rootDsn)
+	// Open a separate connection to the root DSN and create the database if it does not exist
+	initDB, err := sql.Open("mysql", mysql.rootDSN+"?multiStatements=true")
 	if err != nil {
-		log.Fatalln("Error opening MySQL rootDsn -", err)
+		log.Fatalln("Error opening MySQL root DSN -", err)
 	}
-	if err := db.Ping(); err != nil {
-		_ = db.Close()
-		log.Fatalln("Error connecting to MySQL rootDsn -", err)
+	if err = initDB.Ping(); err != nil {
+		if err = initDB.Close(); err != nil {
+			log.Fatalln("Error closing broken MySQL root DSN -", err)
+		}
+		log.Fatalln("Error connecting to MySQL root DSN -", err)
 	}
-
-	// Create the database if it does not exist
-	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s", dbName))
+	_, err = initDB.Exec("CREATE DATABASE IF NOT EXISTS " + dbName)
 	if err != nil {
 		log.Fatalln("Error creating database -", err)
 	}
-	if err := db.Close(); err != nil {
-		log.Fatalln("Error closing database -", err)
-	}
 
-	// Connect to the database
-	db, err = sql.Open("mysql", mysql.rootDsn+dbName+"?multiStatements=true")
+	// Move to the database and run 'init_tables.sql' script.
+	_, err = initDB.Exec("USE " + dbName)
 	if err != nil {
-		log.Fatalln("Error opening database -", err)
-	}
-	if err = db.Ping(); err != nil {
-		_ = db.Close()
-		log.Fatalln("Error connecting to database -", err)
+		log.Fatalln("Error using", dbName, "-", err)
 	}
-
-	// Run 'init.sql' script.
-	_, err = db.Exec(string(initScript))
+	initScript, err := os.ReadFile("internal/database/queries/init_tables.sql")
 	if err != nil {
-		log.Fatalln("Error executing 'init.sql' -", err)
+		log.Fatalln("Error reading init_tables.sql file -", err)
+	}
+	if _, err = initDB.Exec(string(initScript)); err != nil {
+		log.Fatalln("Error executing init_tables.sql -", err)
 	}
-	if err := db.Close(); err != nil {
-		log.Fatalln("Error closing database -", err)
+	if err := initDB.Close(); err != nil {
+		log.Fatalln("Error closing init DB -", err)
 	}
-	return &mysql
+	return
```

Comment:
If it is void, do you still need to return?

Reply:
We actually do! Its a non-void function with named outputs, so it requires the return keyword, but it can imply the values.